src/controller/controller.ts

```diff
@@ -470,16 +470,16 @@ class Controller extends events.EventEmitter {
     private onDeviceLeave(payload: AdapterEvents.DeviceLeavePayload): void {
         debug.log(`Device leave '${payload.ieeeAddr}'`);
 
-        const device = Device.byIeeeAddr(payload.ieeeAddr);
+        const device = payload.ieeeAddr ? Device.byIeeeAddr(payload.ieeeAddr) : Device.byNetworkAddress(payload.networkAddress);
```

Comment:
- Does this only happen on deviceLeave?
- I propose to update the `DeviceLeavePayload` type to: `type DeviceLeavePayload = {networkAddress?: number, ieeeAddr: string} | {networkAddress: number, ieeeAddr?: string}
`

Reply:
- Yes, for now that is the only place it is needed anyway.
- Indeed that would be better. Is it currently an `interface` though.